lib/galaxy/datatypes/media.py

```diff
@@ -0,0 +1,214 @@
+
+    def set_meta(self, dataset, **kwd):
+        metadata, streams = ffprobe(dataset.file_name)
+
+        dataset.metadata.duration = metadata['duration']
+        dataset.metadata.audio_codecs = '|'.join([stream['codec_name'] for stream in streams if stream['codec_type'] == 'audio'])
+        dataset.metadata.sample_rates = '|'.join([stream['sample_rate'] for stream in streams if stream['codec_type'] == 'audio'])
+        dataset.metadata.audio_streams = len([stream for stream in streams if stream['codec_type'] == 'audio'])
+
+
+class Video(Binary):
+
+    MetadataElement(name="resolution_w", default=0, desc="Width of video stream", readonly=True, visible=True, optional=True, no_value=0)
+    MetadataElement(name="resolution_h", default=0, desc="Height of video stream", readonly=True, visible=True, optional=True, no_value=0)
+    MetadataElement(name="fps", default=0, desc="FPS of video stream", readonly=True, visible=True, optional=True, no_value=0)
+    MetadataElement(name="video_codecs", default="", desc="Video codec(s)", readonly=True, visible=True, optional=True, no_value="")
+    MetadataElement(name="audio_codecs", default="", desc="Audio codec(s)", readonly=True, visible=True, optional=True, no_value="")
+    MetadataElement(name="video_streams", default=0, desc="Number of video streams", readonly=True, visible=True, optional=True, no_value=0)
+    MetadataElement(name="audio_streams", default=0, desc="Number of audio streams", readonly=True, visible=True, optional=True, no_value=0)
+
+    def _get_resolution(self, streams):
+        for stream in streams:
+            if stream['codec_type'] == 'video':
+                w = stream['width']
+                h = stream['height']
+                dividend, divisor = stream['avg_frame_rate'].split('/')
+                fps = float(dividend) / float(divisor)
+        else:
+            w = h = fps = 0
+        return w, h, fps
+
+    def set_meta(self, dataset, **kwd):
+        metadata, streams = ffprobe(dataset.file_name)
+        (w, h, fps) = self._get_resolution(streams)
+        dataset.metadata.resolution_w = w
+        dataset.metadata.resolution_h = h
+        dataset.metadata.fps = fps
+
+        dataset.metadata.audio_codecs = '|'.join([stream['codec_name'] for stream in streams if stream['codec_type'] == 'audio'])
+        dataset.metadata.video_codecs = '|'.join([stream['codec_name'] for stream in streams if stream['codec_type'] == 'video'])
+
+        dataset.metadata.audio_streams = len([stream for stream in streams if stream['codec_type'] == 'audio'])
+        dataset.metadata.video_streams = len([stream for stream in streams if stream['codec_type'] == 'video'])
+
+
+class Mkv(Video):
+    file_ext = "mkv"
+
+    def sniff(self, filename):
+        try:
+            metadata, streams = ffprobe(filename)
+            return 'matroska' in metadata['format_name'].split(',')
+        except subprocess.CalledProcessException:
```

Comment:
The try/except is handled by the framework, no need to do this again here. (and all the other places in this PR)

Reply:
Sounds good!

Reply:
> The try/except is handled by the framework, no need to do this again here. (and all the other places in this PR)

For future reference, that's true only for `sniff()` methods, not `set_metadata()`.